tests/acceptance/comparative_booster_study/dataset_definition.py

```diff
@@ -0,0 +1,701 @@
+#      returning="group_6",
+#      use_most_frequent_code=True,
+#      return_expectations={
+#        "category": {"ratios": {"1": 0.2, "2": 0.2, "3": 0.2, "4": 0.2, "5": 0.2}},
+#        "incidence": 0.8,
+#      },
+#    ),
+
+
+#######################################################################################
+# Practice and patient ID variables
+#######################################################################################
+
+practice_reg = practice_registration_as_of(baseline_date)
+
+dataset.practice_id = practice_reg.practice_pseudo_id
+# STP is an NHS administration region based on geography
+dataset.stp = practice_reg.practice_stp
+# NHS administrative region
+dataset.region = practice_reg.practice_nuts1_region_name
+
+address = address_as_of(baseline_date)
+
+# Middle Super Output Area
+dataset.msoa = address.msoa_code
+# Index of Multiple Deprevation Rank (rounded down to nearest 100)
+dataset.imd = address.imd_rounded
+# Rurality
+dataset.rural_urban = address.rural_urban_classification
+
+
+#######################################################################################
+# Occupation / residency
+#######################################################################################
+
+# Health or social care worker
+vaxx_job = schema.occupation_on_covid_vaccine_record
+dataset.hscworker = vaxx_job.take(vaxx_job.is_healthcare_worker).exists_for_patient()
+
+dataset.care_home_type = case(
+    when(
+        address.care_home_is_potential_match
+        & ~address.care_home_requires_nursing
+        & address.care_home_does_not_require_nursing
+    ).then("Carehome"),
+    when(
+        address.care_home_is_potential_match
+        & address.care_home_requires_nursing
+        & ~address.care_home_does_not_require_nursing
+    ).then("Nursinghome"),
+    when(address.care_home_is_potential_match).then("Mixed"),
+)
+
+# Simple care home flag
+dataset.care_home_tpp = dataset.care_home_type.is_not_null()
+
+# Patients in long-stay nursing and residential care
+dataset.care_home_code = has_prior_event(codelists.carehome)
+
+
+#######################################################################################
+# Pre-baseline events where event date is of interest
+#######################################################################################
+
+primary_care_covid_events = events.take(
+    events.ctv3_code.is_in(
+        combine_codelists(
+            codelists.covid_primary_care_code,
+            codelists.covid_primary_care_positive_test,
+            codelists.covid_primary_care_sequelae,
+        )
+    )
+)
+
+dataset.primary_care_covid_case_0_date = (
+    primary_care_covid_events.take(events.date.is_on_or_before(baseline_date))
+    .sort_by(events.date)
+    .last_for_patient()
+    .date
+)
+
+# Covid test dates from SGSS
+covid_tests = schema.sgss_covid_all_tests
+prior_tests = covid_tests.take(covid_tests.specimen_date.is_on_or_before(baseline_date))
```

Comment:
Just dropping by to note that the field name finally decided on the for the [contract](https://github.com/opensafely-core/databuilder/issues/589) was `specimen_taken_date`